pkgs/development/haskell-modules/configuration-ghcjs-9.x.nix

```diff
@@ -0,0 +1,47 @@
+{ pkgs, haskellLib }:
+
+with haskellLib;
+
+pkgs.lib.composeManyExtensions [
+
+  (self: super: {
+    ghcjs-base = super.ghcjs-base_0_8_0_0;
+  })
+
+  (self: super: {
+    ghcjs-dom-javascript = unmarkBroken super.ghcjs-dom-javascript;
+    ghcjs-dom-jsffi = unmarkBroken super.ghcjs-dom-jsffi;
+  })
+
+  # cabal2nix doesn't properly add dependencies conditional on arch(javascript)
+  (self: super: {
+    ghcjs-base = addBuildDepends (with self; [
+      aeson
+      attoparsec
+      dlist
+      hashable
+      primitive
+      scientific
+      unordered-containers
+      vector
+    ]) super.ghcjs-base;
+    ghcjs-dom = addBuildDepend self.ghcjs-dom-javascript super.ghcjs-dom;
+    ghcjs-dom-javascript = addBuildDepend self.ghcjs-base super.ghcjs-dom-javascript;
+    jsaddle = addBuildDepend self.ghcjs-base super.jsaddle;
+    jsaddle-dom = addBuildDepend self.ghcjs-base super.jsaddle-dom;
+  })
+
+  # https://gitlab.haskell.org/ghc/ghc/-/issues/24664
+  # Profiling fails on 9.6/9.8 for QuickCheck and transitively for dependents
+  (self: super: {
+    QuickCheck = disableLibraryProfiling super.QuickCheck;
+
+    aeson = disableLibraryProfiling super.aeson;
+    ghcjs-base = disableLibraryProfiling super.ghcjs-base;
+    ghcjs-dom = disableLibraryProfiling super.ghcjs-dom;
+    ghcjs-dom-javascript = disableLibraryProfiling super.ghcjs-dom-javascript;
+    jsaddle = disableLibraryProfiling super.jsaddle;
+    jsaddle-dom = disableLibraryProfiling super.jsaddle-dom;
+  })
```

Comment:
Is there a way to apply `disableLibraryProfiling` to everything that depends on QuickCheck? 
Do we even want that?

Reply:
Globally disabling library profiling for the js backend would also be a possibility…

Reply:
Alternatively we could try applying https://gitlab.haskell.org/ghc/ghc/-/merge_requests/12463…

Reply:
Seems reasonable. Maybe backporting the patch is too much investment already. 9.10 is around the corner and we probably shouldn't go above and beyond for the dev preview backend necessarily?

Reply:
How does one disable profiling globally? 
Or did you mean piping that entire package set through `builtins.mapAttrs (_: disableLibraryProfiling)` ?